ghga_service_commons/api/mock_router.py

```diff
@@ -0,0 +1,386 @@
+                        description=(
+                            f"Unable to cast '{value}' to {parameter_type} for "
+                            + f"path '{request.url.path}'"
+                        ),
+                        data={
+                            "value": value,
+                            "parameter_type": parameter_type,
+                            "path": request.url.path,
+                        },
+                    ) from err
+            typed_parameters[parameter_name] = value
+
+        # include request itself if needed (e.g. for header or auth info),
+        if "request" in signature_parameters:
+            typed_parameters["request"] = request
+
+        return typed_parameters
+
+    def _parse_url_parameters(
+        self, url: str, endpoint: RegisteredEndpoint
+    ) -> dict[str, str]:
+        """Produce a dict of path var names (keys) and request url values (values).
+
+        This should always match because we will have already performed the match in
+        _get_registered_endpoint.
+        """
+
+        matched_url = re.search(endpoint.url_pattern, url)
+        matched_url = cast(re.Match, matched_url)  # never None, make type checker happy
+        return matched_url.groupdict()
+
+    def _get_registered_endpoint(self, url: str, method: str) -> RegisteredEndpoint:
+        """Match request URL to a registered endpoint's url pattern.
+
+        Iterate through the registered endpoints for the given method.
+        For each registered endpoint, try to match the request's url to the endpoint
+        pattern. Upon matching, return the endpoint object.
+
+        Args:
+            url: The url of the request.
+            method: The method of the request.
+
+        Raises:
+            HttpException:
+                (with status 404) when unable to find a registered endpoint with a
+                matching URL.
+        """
+        for endpoint in self._methods[method]:
+            matched_url = re.search(endpoint.url_pattern, url)
+            if matched_url:
+                return endpoint
+
+        raise HttpException(
+            status_code=404,
+            exception_id="pageNotFound",
+            description=f"No registered path found for url '{url}' and method '{method}'",
+            data={"url": url, "method": method},
+        )
+
+    def _build_loaded_endpoint_function(self, request: httpx.Request) -> partial:
+        """Match a request to the correct endpoint, build typed parameter dictionary,
+        and return loaded partial function.
+        """
+
+        # get endpoint object that corresponds to the request URL
+        endpoint = self._get_registered_endpoint(
+            url=str(request.url), method=request.method
+        )
+
+        # get the parsed string parameters from the url
+        parsed_url_parameters = self._parse_url_parameters(
+            url=str(request.url), endpoint=endpoint
+        )
```

Comment:
```suggestion
        url = str(request.url)

        # get endpoint object that corresponds to the request URL
        endpoint = self._get_registered_endpoint(
            url=url, method=request.method
        )

        # get the parsed string parameters from the url
        parsed_url_parameters = self._parse_url_parameters(
            url=url, endpoint=endpoint
        )
```

Reply:
Fixed in [2ce6daf](https://github.com/ghga-de/ghga-service-commons/pull/18/commits/2ce6daf24f2699069bfb710c1137dc84cbc9d523)